account_payment_partner/models/account_move.py

```diff
@@ -65,6 +66,20 @@ def _compute_payment_mode(self):
                     move.payment_mode_id = move.with_context(
                         force_company=move.company_id.id
                     ).partner_id.customer_payment_mode_id
+                elif (
+                    move.type in ["out_refund", "in_refund"] and move.reversed_entry_id
+                ):
+                    move.payment_mode_id = (
+                        move.reversed_entry_id.payment_mode_id.refund_payment_mode_id
+                    )
+                elif move.type == "out_refund" and not move.reversed_entry_id:
+                    move.payment_mode_id = move.with_context(
+                        force_company=move.company_id.id
+                    ).partner_id.customer_payment_mode_id.refund_payment_mode_id
+                elif move.type == "in_refund" and not move.reversed_entry_id:
+                    move.payment_mode_id = move.with_context(
+                        force_company=move.company_id.id
+                    ).partner_id.supplier_payment_mode_id.refund_payment_mode_id
```

Comment:
You can refactor this code for better reading (not black formatted):
```suggestion
                elif not move.reversed_entry_id:
                    partner = move.with_context(force_company=move.company_id.id).partner_id
                    if move.type == "out_refund"
                        move.payment_mode_id = partner.customer_payment_mode_id.refund_payment_mode_id
                    elif move.type == "in_refund":
                       move.payment_mode_id = partner.supplier_payment_mode_id.refund_payment_mode_id
```